virtualpdu/main.py

```diff
@@ -47,13 +51,24 @@ def main():
             port = int(config.get(pdu, 'listen_port'))
             community = config.get(pdu, 'community')
 
-            pdu_test_harness = pysnmp_handler.SNMPPDUHarness(
+            pdu_threads.append(pysnmp_handler.SNMPPDUHarness(
                 apc_pdu,
                 listen_address,
                 port,
                 community
-            )
-            pdu_test_harness.start()
+            ))
+
+        for t in pdu_threads:
+            t.start()
+
+        try:
+            for t in pdu_threads:
+                while t.isAlive():
+                    t.join(1)
```

Comment:
So that py2 could handle KeyboardInterrupt, otherwise the thread is stuck in the join(). We added a test for it. Perhaps a comment would help future readers.